### PR TITLE
Sphinx doc infrastructure

### PR DESCRIPTION
**Pull request by maximedenes**
This is the infrastructure for porting the doc to Sphinx.

Before considering this PR ready for merge, I need to:
- [X] add a sphinx-doc job to CI
- [x] reorganize the history
- [x] adapt the `install-doc` target

Depends on #6933 [merged] and #6927 [merged].

### CODE REVIEW

**Comment by ejgallego** [comment]
A couple of quick points:
- LICENSE of some files seems to be missing. [fonts?]
- I am not happy with the way `coqtop` is used now [parsing back output colors...]. I do believe that the current implementation will bring a hard to remove technical debt. Instead, we should define a proper interface for the documenter needs. I volunteer to do the OCaml-part if somebody is able to provide me with a spec of what sphinx needs.

File: INSTALL.doc [hunk 1]
```diff
@@ -26,6 +26,12 @@ To produce all the documents, the following tools are needed:
   - convert (ImageMagick)
   - hevea
   - hacha
```
Comment (Zimmi48) [inline INSTALL.doc]: Will all the previous dependencies continue to be needed?

File: configure.ml [hunk 2]
```diff
@@ -107,7 +107,7 @@ let run ?(fatal=true) ?(err=StdErr) prog args =
       let cmd = String.concat " " (prog::args) in
       let exn = match e with Failure s -> s | _ -> Printexc.to_string e in
       let msg = sprintf "Error while running '%s' (%s)" cmd exn in
-      if fatal then die msg else (printf "W: %s\n" msg; "", [])
+      if fatal then die msg else (printf "W: %s\n%!" msg; "", [])
```
Comment (Zimmi48) [inline configure.ml]: What are all these new `%!`. Is this related to this PR?

**Review by Zimmi48 — COMMENTED** [review]
It's a bit unfortunate to completely lose the commit history during the migration. Isn't there a way to have the new files be modified versions of the older files with edits corresponding to changes in formatting?

**Comment by SkySkimmer** [comment]
>Isn't there a way to have the new files be modified versions of the older files with edits corresponding to changes in formatting?

I think you would need to change the files and rename them in separate commits, AFAIK git's rename detection is a heuristic.

File: configure.ml (hunk 2 above)
Comment (ejgallego) [reply configure.ml]: That adds a flush to the formatter, which is usually necessary for printing to succeed. A better strategy would be to define a `cprintf` function that takes care of this:
```
let cprintf fmt = fprintf (fun fmt -> fprintf fmt "@\n%!") fmt
```
so now on every call of `cprintf "foo"`, the newline and flush are automatically inserted.

Comment (maximedenes) [reply configure.ml]: Yep, I need to separate this out from this PR. I added it because some messages wouldn't show up when testing the dependency checks introduced by this PR.

**Comment by Zimmi48** [comment]
> I think you would need to change the files and rename them in separate commits

I would support this solution even if it means having a broken state in between these two commits.

File: INSTALL.doc (hunk 1 above)
Comment (maximedenes) [reply INSTALL.doc]: I don't know. I believe you've worked on other documentation material (tutorial, faq), so you probably know better what is used there.

Comment (Zimmi48) [reply INSTALL.doc]: The FAQ is no more (It is now part of the wiki) and the tutorials should follow the same trajectory. Apart from the reference manual, the only other important part of the documentation is generated by coqdoc from the standard library so it depends what the dependencies of coqdoc are.

**Comment by psteckler** [comment]
> maybe @psteckler can help with the Sphinx/python side?

Yes, I see where the colors are being parsed. It's in `doc/tools/coqrst/coqdomain.py`, class `AnsiColorsParser`. Is there a better way to make the output pretty? By keywords?

**Comment by cpitclaudel** [comment]
>> LICENSE of some files seems to be missing. [fonts?]
> @cpitclaudel probably knows more.

Which files?

**Comment by cpitclaudel** [comment]
> LICENSE of some files seems to be missing. [fonts?]

In fact, all source files are missing a license header AFAICT :)

File: doc/sphinx/preamble.rst [hunk 3]
```diff
@@ -61,3 +56,4 @@
    \newcommand{\oddS}{\textsf{odd\_S}}
    \newcommand{\Prod}{\textsf{prod}}
    \newcommand{\Pair}{\textsf{pair}}
+   \]
```
Comment (cpitclaudel) [inline doc/sphinx/preamble.rst]: I think this change will break LaTeX compilation.

**Comment by ejgallego** [comment]
> Is there a better way to make the output pretty? By keywords?

Any format that is structured and easy to parse by Python would work.

**Comment by psteckler** [comment]
> Any format that is structured and easy to parse by Python would work.

What is your concern with the current setup? The Coq code to output color codes is fragile, somehow? That it's not human-parseable?

It is a bit unstructured: it says "use this color until I tell you otherwise".

**Comment by ejgallego** [comment]
> What is your concern with the current setup? The Coq code to output color codes is fragile, somehow? That it's not human-parseable?

I'd like to have a clear specification of the interface between the documentation system and Coq.

If the interface has to be color information, then I'm OK, but I wouldn't like the whole documentation system to break because someone tweaks the color scheme of the toplevel.

**Comment by psteckler** [comment]
> I'd like to have a clear specification of the interface between the documentation system and Coq.

The Python should accept any text that `coqtop` prints, which may include valid ANSI color codes. The ANSI parser was extended a bit from the one in `sphinx-contribs`, because `coqtop` prints some extended codes (my understanding, at least). We should probably check that the parser has some default action if the ANSI code is unknown.

I'd think that using some new output format for the sake of the docs would only increase the technical debt load.

**Comment by ejgallego** [comment]
> The Python should accept any text that coqtop prints, which may include valid ANSI color codes. The ANSI parser was extended a bit from the one in sphinx-contribs, because coqtop prints some extended codes (my understanding, at least). We should probably check that the parser has some default action if the ANSI code is unknown.

This doesn't answer my question. Let me rephrase it, why does the document generator need to call into `coqtop`? (in the first place)

> I'd think that using some new output format for the sake of the docs would only increase the technical debt load.

Umm, I'm not sure I can follow your reasoning here, why?

**Comment by cpitclaudel** [comment]
There's plenty of input-output examples in the manual - that's why we run coqtop. It's a python version of coqtex, essentially. The ANSI color parsing could be dropped, of course.

**Comment by ejgallego** [comment]
> There's plenty of input-output examples in the manual - that's why we run coqtop. It's a python version of coqtex, essentially. The ANSI color parsing could be dropped, of course.

I am not familiar with `coqtex`. So given a snippet "foo", what do you usually need to retrieve? The goals? Information about types?

**Comment by cpitclaudel** [comment]
Coqtex takes a latex document, extracts blocks of coq code, runs them, and prints the output. That's how the examples in the manual are generated.

**Comment by ejgallego** [comment]
I see, thanks. I had a closer look and I think the current approach should be OK, given that all that you want is to print Coq's output verbatim. I think parsing ANSI is also fine if you wanna do it.

Maybe `coqtop` could be tweaked to make the file of interaction module easier, but that could be addressed in a separate PR. Excessive goal printing can also be a problem sometimes but `-quiet` should work more reliably in 8.8 if you need it.

File: configure.ml (hunk 2 above)
Comment (maximedenes) [reply configure.ml]: @ejgallego your `cprintf` function doesn't seem to typecheck. What did you mean there?

Comment (maximedenes) [reply configure.ml]: @ejgallego ping

**Comment by maximedenes** [comment]
> I think you would need to change the files and rename them in separate commits, AFAIK git's rename detection is a heuristic.

I'll try that, but I'm afraid the linter is going to not like it.

File: INSTALL.doc (hunk 1 above)
Comment (maximedenes) [reply INSTALL.doc]: Who can tell me more about this?

**Comment by Zimmi48** [comment]
> I'll try that, but I'm afraid the linter is going to not like it.

When the reason is good, you can be granted a linter exception ;-)
For instance, I think it is much more important to keep the history of these files as much as possible than having whitespace-free commits. I'm pretty sure @SkySkimmer would agree.

**Comment by SkySkimmer** [comment]
Of course.

Gaëtan Gilbert

**Comment by maximedenes** [comment]
Sure, but it is not just about getting an exception, it is also about convincing the whitespace hook not to mess with those commits. Hopefully @SkySkimmer provided some instructions.

**Comment by SkySkimmer** [comment]
>Sure, but it is not just about getting an exception, it is also about convincing the whitespace hook not to mess with those commits. Hopefully @SkySkimmer provided some instructions.

It's just a bash script, you can add `exit 0` at the beginning, erase it, `chmod -x .git/hooks/pre-commit` or any other solution.
Or use `git commit --no-verify`.

**Comment by maximedenes** [comment]
@cpitclaudel what is the license of the `UbuntuMono-Square.ttf` file?

**Comment by ejgallego** [comment]
> @cpitclaudel what is the license of the UbuntuMono-Square.ttf file?

They are under the "UBUNTU FONT LICENCE Version 1.0"

> Maybe we don't need to include it, tough. And have it loaded from a CDN.

This is no-go, you cannot break all non-connected builds, in particular build containers may be internet-isolated.

**Comment by maximedenes** [comment]
> This is no-go, you cannot break all non-connected builds, in particular build containers may be internet-isolated.

?? Why would the builds need the font?

**Comment by Zimmi48** [comment]
And why do we need this particular font at all BTW?

**Comment by Zimmi48** [comment]
And I'll remind that @aspiwack has several time expressed his hatred of the default Sphinx theme so I hope that's not the one that has been chosen in the end...

**Comment by maximedenes** [comment]
> And why do we need this particular font at all BTW?

https://github.com/cpitclaudel/coq-rst/commit/d3a17f29befa280c8b707ce643dc6cfc6ecca652

**Comment by maximedenes** [comment]
> And I'll remind that @aspiwack has several time expressed his hatred of the default Sphinx theme so I hope that's not the one that has been chosen in the end...

Maybe the default theme changed, but we did choose the one he didn't like. I like it, and many others do, so I don't see the problem.

On the longer term, we are going to want a unified theme for the website and the manual, though, so we will probably change.

**Comment by ejgallego** [comment]
> Why would the builds need the font?

As to do sanity / installs check, just to name one of the many possibilities.

**Comment by maximedenes** [comment]
Where should I mention the license for this file, then?

**Comment by maximedenes** [comment]
> As to do sanity / installs check, just to name one of the many possibilities.

After all, I'd like to know what you mean exactly. What kind of checks?

**Comment by Zimmi48** [comment]
> Where should I mention the license for this file, then?

I suggest you add a paragraph in the beginning of the `doc/LICENSE` file.

**Comment by maximedenes** [comment]
> I suggest you add a paragraph in the beginning of the doc/LICENSE file.

Will do once I get more information on license status of other files from @cpitclaudel.

File: configure.ml (hunk 2 above)
Comment (ejgallego) [reply configure.ml]: Sorry there was a typo it should be:
```ocaml
let cprintf fmt = kfprintf (fun fmt -> fprintf fmt "@\n%!") fmt
```

Comment (ejgallego) [reply configure.ml]: Note the eta expansion is usually mandatory.

**Comment by ejgallego** [comment]
> I don't think the checks or install targets are going to enumerate files anyway, but I don't want to fight.

Depends how good your packager is. But having documentation that depends on online resources is going to create many problems for distros, etc...

**Comment by cpitclaudel** [comment]
> Will do once I get more information on license status of other files from @cpitclaudel.

That font is a modified version of the standard Ubuntu fonts, added mostly for cosmetic reasons (to get centered `,`, `?`, and `.`, because CSS is bad at this).  Removing it shouldn't be a huge issue.

All of the Python files should have the standard copyright headers with my name and the year, and LGPL (I think that's what Coq is using, right?)

Are there other files?

**Comment by Zimmi48** [comment]
Coq is using LGPL v2.1 except for its documentation where it's using OPL. That's said it could make sense to put all the new infrastructure file under the LGPL license as well rather than the OPL.

**Comment by maximedenes** [comment]
> with my name and the year

@Zimmi48 That's going to go to CREDITS rather than the headers, right?

**Comment by Zimmi48** [comment]
> That's going to go to CREDITS rather than the headers, right?

That would be preferable if @cpitclaudel doesn't mind.

**Comment by cpitclaudel** [comment]
> > That's going to go to CREDITS rather than the headers, right?

> That would be preferable if @cpitclaudel doesn't mind.

I don't mind, but I don't think that's HOW the GPL works: https://www.gnu.org/licenses/gpl-howto.en.html ; you need a copyright and a notice license in each file.

**Comment by Zimmi48** [comment]
Yep @cpitclaudel but we have standard headers. See `dev/header`. So we fulfill these conditions of (L)GPL.

**Comment by maximedenes** [comment]
> Yep @cpitclaudel but we have standard headers. See dev/header. So we fulfill these conditions of (L)GPL.

I'm not sure I understand this answer. Concretely, who's going to be the copyright holder of the files contained in this PR?

**Comment by maximedenes** [comment]
(but ok, meanwhile I'll add the standard headers)

**Comment by cpitclaudel** [comment]
>> Yep @cpitclaudel but we have standard headers. See dev/header. So we fulfill these conditions of (L)GPL.

> I'm not sure I understand this answer. Concretely, who's going to be the copyright holder of the files contained in this PR?

Me, unless you make me sign a copyright assignment.

**Comment by Zimmi48** [comment]
> do we have versions of those headers for other languages than OCaml

You can reuse the ones in the Makefile.

> Concretely, who's going to be the copyright holder of the files contained in this PR?

Clément of course.

> Except these headers state that the copyright is help by a bunch of labs, none of which I'm affiliated with.

Obviously, you looked at a file in an outdated branch. Check https://github.com/coq/coq/blob/master/dev/header which contains the mention "and contributors" (and also a reference to the CREDITS file).

**Comment by Zimmi48** [comment]
> who's going to be the copyright holder of the files contained in this PR?

Basic copyright law reminder: whatever we do with the header, even if we fuck up badly, that doesn't change who's the copyright holder. That only makes the lawyers' life more difficult.

File: configure.ml (hunk 2 above)
Comment (maximedenes) [reply configure.ml]: @ejgallego I'm still a bit lost. Why is the output channel called `fmt`?

Comment (maximedenes) [reply configure.ml]: A mistake, I guess, so the eta-expansion is not needed.

Comment (maximedenes) [reply configure.ml]: Oh no sorry, I see now. I was confused about a formatter for an output channel vs the output channel itself. Thanks, will split this part out of this PR.

Comment (ejgallego) [reply configure.ml]: > @ejgallego I'm still a bit lost. Why is the output channel called fmt?

It is not an output channel but a formatter.

> A mistake, I guess, so the eta-expansion is not needed.

?? Eta expansion is needed if you want your code to work in older OCamls.

Comment (maximedenes) [reply configure.ml]: I understand the quiproquo now. `configure.ml` uses `Printf`, not `Format`. In particular `Printf.kfprintf` takes an `output_channel`, not a `formatter`.

Comment (ejgallego) [reply configure.ml]: Oh, is it still using `Printf`? I suggest you switch it to use `Format`.

**Comment by cpitclaudel** [comment]
> Obviously, you looked at a file in an outdated branch. 

Indeed, in the latest release.  Thanks.

File: configure.ml (hunk 2 above)
Comment (maximedenes) [reply configure.ml]: I'll leave that for another time, it is critical to get this sphinx infra PR ready.

**Comment by Zimmi48** [comment]
@cpitclaudel So you're OK with the generic headers for your files?

**Comment by cpitclaudel** [comment]
Sure

**Comment by maximedenes** [comment]
@cpitclaudel what is this `hint.min.css` file that contains HTML?

**Comment by maximedenes** [comment]
I've turned this PR into pure infrastructure: it produces an empty HTML manual, with the old LaTeX one still complete. This way, I can take care of history preservation for chapters in separate PRs.

**Comment by cpitclaudel** [comment]
@cpitclaudel what is this hint.min.css file that contains HTML?

A temporary file that shouldn't have been checked in, I expect.  Woops.

**Comment by maximedenes** [comment]
Well, it is referenced from `doc/tools/coqrst/coqdomain.py`, so I'm not sure.

**Comment by cpitclaudel** [comment]
Ah, good point. It was supposed to be https://raw.githubusercontent.com/chinchang/hint.css/master/hint.min.css.
This is a tooltip library, the intent of which was to display a small label when hovering on a repetition annotation in a tactic notation.

**Comment by maximedenes** [comment]
> Ah, good point. It was supposed to be https://raw.githubusercontent.com/chinchang/hint.css/master/hint.min.css.
> This is a tooltip library, the intent of which was to display a small label when hovering on a repetition annotation in a tactic notation.

@cpitclaudel That file is licensed under AGPL3, so I don't think we can use it (maybe I'm wrong). Do you think I can safely remove the file and the line that mentions it?

**Comment by maximedenes** [comment]
Also, is there other external code that was integrated, that I should check?

**Comment by cpitclaudel** [comment]
> Do you think I can safely remove the file and the line that mentions it?

Yes, definitely; it's a small cosmetic thing.  (It would be nice to scrub the history, too). Sorry for the trouble.

File: doc/LICENSE [hunk 4]
```diff
@@ -2,11 +2,14 @@ The Coq Reference Manual is a collective work from the Coq Development
 Team whose members are listed in the file CREDITS of the Coq source
 package. All related documents (the LaTeX and BibTeX sources, the
 embedded png files, and the PostScript, PDF and html outputs) are
-copyright (c) INRIA 1999-2006. The material connected to the Reference
-Manual may be distributed only subject to the terms and conditions set
-forth in the Open Publication License, v1.0 or later (the latest
-version is presently available at http://www.opencontent.org/openpub/).
-Options A and B are *not* elected.
+copyright (c) INRIA 1999-2006, with the exception of the Ubuntu font files
+(UbuntuMono-Square.ttf and UbuntuMono-B.ttf) which are Copyright 2010,2011
```
Comment (cpitclaudel) [inline doc/LICENSE]: UbuntuMono-Square.ttf is actually generated from UbuntuMono-Regular.

File: doc/sphinx/conf.py [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+##########################################################################
+##         #   The Coq Proof Assistant / The Coq Development Team       ##
+##  v      #   INRIA, CNRS and contributors - Copyright 1999-2018       ##
+## <O___,, #       (see CREDITS file for the list of authors)           ##
+##   \VV/  ###############################################################
+##    //   #    This file is distributed under the terms of the         ##
+##         #     GNU Lesser General Public License Version 2.1          ##
+##         #     (see LICENSE file for the text of the license)         ##
+##########################################################################
+# -*- coding: utf-8 -*-
```
Comment (cpitclaudel) [inline doc/sphinx/conf.py]: That coding line should be above the license header

File: doc/LICENSE (hunk 4 above)
Comment (maximedenes) [reply doc/LICENSE]: So should I update the text? If yes, what should I put/remove?

File: doc/sphinx/conf.py (hunk 5 above)
Comment (maximedenes) [reply doc/sphinx/conf.py]: Thanks, will fix! Out of curiosity, what uses it?

File: doc/tools/coqrst/notations/TacticNotationsLexer.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
+# Generated from TacticNotations.g by ANTLR 4.7
```
Comment (cpitclaudel) [inline doc/tools/coqrst/notations/TacticNotationsLexer.py]: Should this be checked into the repo?

File: doc/tools/coqrst/notations/html.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
+##########################################################################
+##         #   The Coq Proof Assistant / The Coq Development Team       ##
+##  v      #   INRIA, CNRS and contributors - Copyright 1999-2018       ##
+## <O___,, #       (see CREDITS file for the list of authors)           ##
+##   \VV/  ###############################################################
+##    //   #    This file is distributed under the terms of the         ##
+##         #     GNU Lesser General Public License Version 2.1          ##
+##         #     (see LICENSE file for the text of the license)         ##
+##########################################################################
+"""A visitor for ANDTLR notation ASTs, producing raw HTML.
```
Comment (cpitclaudel) [inline doc/tools/coqrst/notations/html.py]: woops, there's a typo here ("ANDTLR")

File: doc/sphinx/conf.py (hunk 5 above)
Comment (cpitclaudel) [reply doc/sphinx/conf.py]: Python and Emacs

File: doc/tools/coqrst/notations/TacticNotationsLexer.py (hunk 6 above)
Comment (maximedenes) [reply doc/tools/coqrst/notations/TacticNotationsLexer.py]: I've been wondering. It was checked in your repo, and it reduces the number of dependencies. So probably convenient during the migration period, but on the longer term, we'll want to build everything from sources, I guess.

File: doc/LICENSE (hunk 4 above)
Comment (cpitclaudel) [reply doc/LICENSE]: Unclear; I just meant to point it out.  Maybe "derived from file Copyright …"

File: doc/tools/coqrst/notations/html.py (hunk 7 above)
Comment (maximedenes) [reply doc/tools/coqrst/notations/html.py]: There seem to be 4 occurrences of the typo :)

**Comment by cpitclaudel** [comment]
> Also, is there other external code that was integrated, that I should check?

I just went through everything.  I don't think so.

**Comment by maximedenes** [comment]
Success here: https://gitlab.com/maximedenes/coq/pipelines/18647660 (modulo some network issues that gilab had).

File: default.nix [hunk 8]
```diff
@@ -43,6 +43,15 @@ stdenv.mkDerivation rec {
     # CoqIDE dependencies
     ocamlPackages.lablgtk
 
+  ] else []) ++ (if buildDoc then [
+
+    # Sphinx doc dependencies
+    ocamlPackages.lablgtk
```
Comment (Zimmi48) [inline default.nix]: Is it because CoqIDE is built whenever the documentation is built? We should remember to fix that.

File: default.nix [hunk 9]
```diff
@@ -43,6 +43,15 @@ stdenv.mkDerivation rec {
...
+    pkgconfig (python3.withPackages
```
Comment (Zimmi48) [inline default.nix]: Could you put this on a new line for more clarity? Otherwise, it may look like a function application to the untrained eye.

File: default.nix [hunk 10]
```diff
@@ -43,6 +43,15 @@ stdenv.mkDerivation rec {
...
+      (ps: [ ps.sphinx ps.sphinx_rtd_theme ps.pexpect ps.beautifulsoup4
+             ps.antlr4-python3-runtime ps.sphinxcontrib-bibtex ]))
+     antlr4
```
Comment (Zimmi48) [inline default.nix]: Extra-space.

File: doc/sphinx/MIGRATING [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+  branch. If your commit is recent, you should see a link on your
+  fork's code page to do that. Otherwise, you may need to go to your
+  branch on GitHub to do that.
+
+# Issues/Questions/Suggestions
+
+  As the migration proceeds, if you have technical issues, have a more general question, or want to suggest something, please contact:
+
+    Paul Steckler <steck@stecksoft.com>
+    Maxime Dénès <maxime.denes@inria.fr>
```
Comment (Zimmi48) [inline doc/sphinx/MIGRATING]: Do you want to keep this file, including this, and the reference above to your repository?

File: doc/sphinx/conf.py [hunk 12]
```diff
@@ -0,0 +1,400 @@
...
+
+# The encoding of source files.
+#source_encoding = 'utf-8-sig'
+
+# The master toctree document.
+master_doc = 'index'
+
+# General information about the project.
+project = 'Coq'
+copyright = '2016, Inria'
```
Comment (Zimmi48) [inline doc/sphinx/conf.py]: Ouch!

File: doc/sphinx/conf.py [hunk 13]
```diff
@@ -0,0 +1,400 @@
...
+author = 'The Coq Development Team'
+
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short X.Y version.
+version = '8.7'
+# The full version, including alpha/beta/rc tags.
+release = '8.7.dev'
```
Comment (Zimmi48) [inline doc/sphinx/conf.py]: Re-ouch!

File: doc/sphinx/conf.py [hunk 14]
```diff
@@ -0,0 +1,400 @@
...
+# documentation.
+#html_theme_options = {}
+
+# Add any paths that contain custom themes here, relative to this directory.
+import sphinx_rtd_theme
+html_theme_path = [sphinx_rtd_theme.get_html_theme_path()]
+
+# The name for this set of Sphinx documents.
+# "<project> v<release> documentation" by default.
+#html_title = 'Coq 8.5 v8.5pl1'
```
Comment (Zimmi48) [inline doc/sphinx/conf.py]: Could you remove the commented out lines?

File: doc/sphinx/conf.py [hunk 15]
```diff
@@ -0,0 +1,400 @@
...
+#html_extra_path = []
+
+# If not None, a 'Last updated on:' timestamp is inserted at every page
+# bottom, using the given strftime format.
+# The empty string is equivalent to '%b %d, %Y'.
+#html_last_updated_fmt = None
+
+# If true, SmartyPants will be used to convert quotes and dashes to
+# typographically correct entities.
+html_use_smartypants = False # FIXME wrap code in <code> tags, otherwise quotesget converted in there
```
Comment (Zimmi48) [inline doc/sphinx/conf.py]: There's a FIXME here, but I suppose this can be done later on.

File: doc/sphinx/conf.py [hunk 16]
```diff
@@ -0,0 +1,400 @@
...
+# -- Options for Texinfo output -------------------------------------------
+
+# Grouping the document tree into Texinfo files. List of tuples
+# (source start file, target name, title, author,
+#  dir menu entry, description, category)
+texinfo_documents = [
+    (master_doc, 'Coq85', 'Coq 8.5 Documentation',
+     author, 'Coq85', 'One line description of project.',
+     'Miscellaneous'),
+]
```
Comment (Zimmi48) [inline doc/sphinx/conf.py]: Re-re-ouch!

File: doc/sphinx/conf.py [hunk 17]
```diff
@@ -0,0 +1,400 @@
...
+# 'ja' uses this config value.
+# 'zh' user can custom change `jieba` dictionary path.
+#html_search_options = {'type': 'default'}
+
+# The name of a javascript file (relative to the configuration directory) that
+# implements a search results scorer. If empty, the default will be used.
+#html_search_scorer = 'scorer.js'
+
+# Output file base name for HTML help builder.
+htmlhelp_basename = 'Coq85doc'
```
Comment (Zimmi48) [inline doc/sphinx/conf.py]: What's that?

File: doc/sphinx/conf.py [hunk 18]
```diff
@@ -0,0 +1,400 @@
...
+
+latex_additional_files = ["_static/coqnotations.sty"]
+
+# Grouping the document tree into LaTeX files. List of tuples
+# (source start file, target name, title,
+#  author, documentclass [howto, manual, or own class]).
+latex_documents = [
+    (master_doc, 'Coq85.tex', 'Coq 8.5 Documentation',
+     'The Coq Development Team (edited by C. Pit-Claudel)', 'manual'),
+]
```
Comment (Zimmi48) [inline doc/sphinx/conf.py]: Please update.

File: doc/sphinx/conf.py [hunk 19]
```diff
@@ -0,0 +1,400 @@
...
+
+
+# -- Options for manual page output ---------------------------------------
+
+# One entry per manual page. List of tuples
+# (source start file, name, description, authors, manual section).
+man_pages = [
+    (master_doc, 'coq85', 'Coq 8.5 Documentation',
+     [author], 1)
+]
```
Comment (Zimmi48) [inline doc/sphinx/conf.py]: Could we avoid as much as possible to hardcode the version number everywhere? That's a recipe for disaster.

File: doc/sphinx/conf.py [hunk 20]
```diff
@@ -0,0 +1,400 @@
...
+#texinfo_no_detailmenu = False
+
+
+# -- Options for Epub output ----------------------------------------------
+
+# Bibliographic Dublin Core info.
+epub_title = project
+epub_author = author
+epub_publisher = author
+epub_copyright = copyright
```
Comment (Zimmi48) [inline doc/sphinx/conf.py]: If this is useful this should be fixed; if not, probably cleaned up.

File: doc/sphinx/coqdoc.css [hunk 21]
```diff
@@ -0,0 +1,338 @@
...
+             font-size : 115% }
+
+h4.section {
+/*
+                   background-color: rgb(80%,80%,80%);
+                   max-width: 20em;
+                   padding-left: 5px;
+                   padding-top: 5px;
+                   padding-bottom: 5px;
+*/
```
Comment (Zimmi48) [inline doc/sphinx/coqdoc.css]: Could the commented out lines be removed?

File: doc/sphinx/coqdoc.css [hunk 22]
```diff
@@ -0,0 +1,338 @@
...
+.id[title="class"] {
+    color: rgb(0%,0%,80%);
+}
+
+.id[title="keyword"] {
+ color : #cf1d1d;
+/*     color: black; */
+}
+
+/* Deprecated rules using the 'type' attribute of <span> (not xhtml valid) */
```
Comment (Zimmi48) [inline doc/sphinx/coqdoc.css]: Is this a problem?

**Review by Zimmi48 — COMMENTED** [review]
Oops, my review comes a bit late:

Thanks @maximedenes for the update to `default.nix`. In fact, I didn't initially add a `buildDoc` argument precisely because I was waiting for the migration to Sphinx.

Note that you should also update the build target when `buildDoc` is set to `true` (because, contrary to CoqIDE, it doesn't build by default when the dependencies are found).

And BTW thanks @cpitclaudel for the very impressive work.

**Comment by maximedenes** [comment]
@Zimmi48 Sorry for the race condition, I thought everything had been reviewed. Most of the outdated info you pointed out is used by other backends than the one this PR uses. But indeed a cleanup should happen at some point.

**Comment by Zimmi48** [comment]
Good. Nothing urgent though!

File: doc/sphinx/conf.py (hunk 17 above)
Comment (cpitclaudel) [reply doc/sphinx/conf.py]: That's if you want to build a .chm version of the Coq manual; it can be removed, I think :)

File: doc/sphinx/conf.py (hunk 20 above)
Comment (cpitclaudel) [reply doc/sphinx/conf.py]: I think an epub version of the manual isn't the top priority, but it might be nice? :)

File: doc/sphinx/coqdoc.css (hunk 21 above)
Comment (cpitclaudel) [reply doc/sphinx/coqdoc.css]: Definitely

File: doc/sphinx/coqdoc.css (hunk 22 above)
Comment (cpitclaudel) [reply doc/sphinx/coqdoc.css]: This file was copied from Coqdoc, I think

**Comment by cpitclaudel** [comment]
> And BTW thanks @cpitclaudel for the very impressive work.

Thanks everyone for bringing this to completion!

**Comment by aspiwack** [comment]
And congratulations to everybody. This is definitely worth it! And we're only just beginning to reap the benefits.

PS: changethattheme changethattheme changethattheme changethattheme changethattheme changethattheme changethattheme changethattheme.

**Comment by ejgallego** [comment]
@aspiwack why don't you provide a PR making the theme user-selectable, and a theme of your liking?

**Comment by maximedenes** [comment]
> why don't you provide a PR making the theme user-selectable, and a theme of your liking?

I would support that!

**Comment by cpitclaudel** [comment]
> > > with my name and the year

> > That's going to go to CREDITS rather than the headers, right?

> That would be preferable if @cpitclaudel doesn't mind. [...]

Bonus point if it does happen before the release :) Am I missing something?

**Comment by maximedenes** [comment]
Hi @cpitclaudel !

Sorry, I should have updated `CREDITS` already, that's an oversight from my part. I was really too overloaded with this release, I should have delegated the integration of the Sphinx manual.

About other credits (`CHANGES`, and probably a specific section of the manual), I had written this in the release notes:

> The integration of the reference manual in our new Sphinx-based infrastructure is still ongoing. The current status can be seen online and will be updated as we integrate new chapters. All contributors to the porting effort will be credited once the integration is done (before the 8.8.0 release).

So sorry again for the intermediate state of this beta, but I'll make sure everything is in order (and credits given) before 8.8.0. I hope that's ok.

**Comment by cpitclaudel** [comment]
No problem at all. Thanks a lot!
